app/controllers/users_controller.rb

```diff
@@ -1,58 +1,76 @@
+  def edit
+    @organizations = unclaimed_organizations
+    @organizations.push @user.organization if @user.organization.present?
+    # the page is passed in by application.html.erb
+    # via <%= link_to "Edit Profile", edit_user_path(user_id: current_user.id, page: "password_reset") %>
+    render "edit"
+  end
 
- end
+  def create
+    # we generate the password_digest automatically
+    # has_secure_password make sure password == password_confirmation
+    user_params = params.require(:user).permit(:first_name, :last_name, :email, :password, :password_confirmation)
+    @user = User.new user_params
 
- def create
-   # we generate the password_digest automatically
-   # has_secure_password make sure password == password_confirmation
-   user_params = params.require(:user).permit(:first_name, :last_name, :email, :password, :password_confirmation)
-   @user = User.new user_params
-   if @user.save
-     session[:user_id] = @user.id
-     #redirect_to root_path, notice: "Account Created!"
-     @user.generate_account_verification_data
-     AccountVerificationsMailer.send_verification_instructions(@user).deliver_later
-     render "users/account_verifications/create"
+    if @user.save
+      session[:user_id] = @user.id
+      #redirect_to root_path, notice: "Account Created!"
+      @user.generate_account_verification_data
+      AccountVerificationsMailer.send_verification_instructions(@user).deliver_later
+      render "users/account_verifications/create"
+    else
+      render :new
+    end
+  end
 
-   else
-     render :new
-   end
 
- end
+  def update
+    if @user.update update_user_params
+      redirect_to admin_users_path, notice: "Account Updated"
+    else
+      render :edit
+    end
+  end
 
 
- def update
-   # puts params
+  def destroy
+    if current_user == @user
+      session[:user_id] = nil
+      path = root_path
+    else
+      path = admin_users_path
+    end
+    @user.destroy
+    redirect_to path, notice: "User deleted!"
+  end
 
-   @user = User.find_by_id params[:id]
+private
 
-   user_params = params.require(:user).permit(:first_name, :last_name, :email, :password, :password_confirmation)
+  def find_user
+    @user ||= User.find_by_id params[:id]
+  end
 
-   if @user.update user_params
-     redirect_to root_path, notice: "Account Updated"
-   else
-     render :edit
-   end
+  def update_user_params
+    if can? :manage, @user
+      params.require(:user).permit(:first_name, :last_name, :email, :password, :password_confirmation, :admin, :organization_id)
+    else
+      params.require(:user).permit(:first_name, :last_name, :email, :password, :password_confirmation)
+    end
+  end
 
- end
-
-
- def destroy
-   user = User.find_by_id params[:id]
-   session[:user_id] = nil
-   user.destroy
-   redirect_to root_path, notice: "User deleted!"
- end
+  def unclaimed_organizations
+    unclaimed = []
+    Organization.all.each do |org|
+      unclaimed.push(org) if org.user.nil?
+    end
+    unclaimed
```

Comment:
This is the work of the model. Build a method in `Organization` model that may look like:
```ruby
def self.unclaimed_organizations
  Organization.where(user_id: nil)
end
```